client/webpack-hot-middleware-client.js

```diff
@@ -41,7 +41,8 @@ webpackHotMiddlewareClient.subscribe((obj) => {
   const fn = handlers[obj.action]
   if (fn) {
     const data = obj.data || []
-    fn(...data)
+    const route = data[0].replace(/[index]*\.json$/, '')
```

Comment:
Please change `server/hot-reloader.js` instead of this file. Maybe `toRoute` method.